experimental/webgpu/simple_allocator.c

```diff
@@ -180,13 +180,12 @@ static iree_status_t iree_hal_webgpu_simple_allocator_allocate_buffer(
     usage_flags |= WGPUBufferUsage_Indirect;
   }
 
-  const bool has_initial_data = !iree_const_byte_span_is_empty(initial_data);
   WGPUBufferDescriptor descriptor = {
       .nextInChain = NULL,
       .label = NULL,
       .usage = usage_flags,
       .size = allocation_size,
-      .mappedAtCreation = has_initial_data,
+      .mappedAtCreation = false,
```

Comment:
> This likely breaks WebGPU but we can fix that with better solutions -
the existing approach worked a bit but ran down code paths that were
dangerously close to blocking and it didn't account for a lot of buffer
types.

Tested simple_abs, multiple_results, and fullyconnected. All still work as they did before, though those programs are pretty simple and might not be going down this path anyways.

Reply:
whew - was worried about that - I think I added initial_data for webgpu in particular but it's just too costly to have as part of the core API - if we can live without it that's great.

Reply:
(we can have special paths for file transfer on webgpu that use pre-mapped staging buffers and stuff if we needed, or issued queue writeBuffer ops which would be even better)